DatadogSessionReplay/Sources/Processor/Processor.swift

```diff
@@ -63,6 +69,10 @@ internal class Processor: Processing {
             .compactMap { node in node.semantics.wireframesBuilder }
             .flatMap { nodeBuilder in nodeBuilder.buildWireframes(with: wireframesBuilder) }
 
+        #if DEBUG
+        interceptWireframes?(wireframes)
+        #endif
```

Comment:
Here we hook into SR implementation to intercept `[SRWireframe]` for creating snapshot images. It could be done in more excellent way by refactoring `Processor` implementation and adding more abstractions - e.g. pipeline-like architecture: `flatten(snapshot) -> nodes` → `map(nodes) -> wireframes` → `export(wireframes)`, and injecting mock implementation of the last phase. Given that current hook is very minimal and non-intrusive for `Release` builds, I'd rather keep it for now and only think of sth more elegant if we find more reason for such refactoring.

Reply:
Is `DEBUG` compiler directive is the right one? can't it be `TEST` or a custom one?
not a big fan of test-specific code bit your comment makes total sense 👍

Reply:
Anything other than default (`Debug` | `Release`) would be hard to configure. The SR is linked to `SRExample` as SPM package, so that flag would have to be defined in `Package.swift` AFAIK 😞. I'd rather search for architecture-level solution (more abstraction like explained ☝️), but IMO this is too early for it. WDYT?